SIGL-SIWEB/new-intra
# Notes grades

## Added
### teacher-grade-component
The front gets the teacher's courses by his login, the list of students, and the grade of each students in these courses.

The component uses Material Table to display the grades, and FormGroup/FormControl to keep track of forms updates.

## Modified
### types
New types for StudentGrades

### notes-component
button to display the teacher-grade-component

removed old grades API calls

### notes-service
new put request to back

## Additional Info
In order to test this component: 
- You need to use the grades-put branch of the back if it has not been merged yet

You can execute the following commands on localhost:3000/docs/swagger:
- If not done yet, you should get /init on back end to initialize the student list or create some students
- Create a teacher with your login (post /teachers)
- Create at least one Course (post /courses)
- Assign at least a Course to the teacher (/teachers/{teacherid}/addCourse)

You can then go to localhost:4200/notes to try it out
